frontend/catalyst/pennylane_extensions.py

```diff
@@ -334,6 +334,8 @@ def _decomp_to_unitary(self, *_args, **_kwargs):
         overriden_methods = [  # pragma: no cover
             (qml.ops.Controlled, "has_decomposition", lambda self: True),
             (qml.ops.Controlled, "decomposition", _decomp_to_unitary),
+            (qml.ops.BlockEncode, "has_decomposition", lambda self: True),
+            (qml.ops.BlockEncode, "decomposition", _decomp_to_unitary),
```

Comment:
minor, but would it be more maintainable to add this to the TOML?

Reply:
I'll update it. thanks Josh :)

Reply:
Great idea 👍

Reply:
Actually this will not be possible unless the device in question adds the operator to it's list of "supported ops" (in it's PL device class).

Reply:
Should we solve this on the lightning side instead? It won't really be useable until next release though.

Reply:
@josh146 another idea is that we implement this generically for all gates, that is if a gate doesn't define a decomposition, Catalyst could check if it defines a matrix and patch its decomposition towards that 🤔 (it could fit into this story: https://app.shortcut.com/xanaduai/epic/53805?group_by=none&vc_group_by=day&ct_workflow=all&cf_workflow=500000005)

Reply:
I'm in favour of that :+1: I think it would be important to somehow know if this is happening/deactivate this behaviour if required, because there are likely cases (e.g., benchmarking) where we might want to ensure this is not happening?

Reply:
Hmm, yeah maybe you are right. I mean the state right now is that `BlockEncode` is simply not supported on lightning (whether you use Catalyst or not). That begs question why Catalyst should introduce this behaviour, and whether it wouldn't be better implemented in PennyLane. But I'm sure they will have some reasons against this

Reply:
>That begs question why Catalyst should introduce this behaviour, and whether it wouldn't be better implemented in PennyLane

Yep good point. I was thinking in my comment above explicitly for QSVT benchmarking

Reply:
We should be able to the logic

Reply:
A PR is now available in lightning: https://github.com/PennyLaneAI/pennylane-lightning/pull/599